server/base/src/main/java/org/apache/accumulo/server/util/RpcWrapper.java

```diff
@@ -36,20 +42,51 @@
  * @since 1.6.1
  */
 public class RpcWrapper {
+  private static final Logger log = LoggerFactory.getLogger(RpcWrapper.class);
+
+  public static <T> T service(final T instance, @SuppressWarnings("rawtypes") final Map<String,ProcessFunction<T,? extends TBase>> processorView) {
+    // Get a handle on the isOnewayMethod and make it accessible
+    final Method isOnewayMethod;
+    try {
+      isOnewayMethod = ProcessFunction.class.getDeclaredMethod("isOneway");
+    } catch (NoSuchMethodException e) {
+      throw new RuntimeException("Could not access isOneway method", e);
+    } catch (SecurityException e) {
+      throw new RuntimeException("Could not access isOneway method", e);
+    }
+    isOnewayMethod.setAccessible(true);
+
+    final Set<String> onewayMethods = new HashSet<String>();
+    for (@SuppressWarnings("rawtypes") Entry<String,ProcessFunction<T,? extends TBase>> entry : processorView.entrySet()) {
+      try {
+        if ((Boolean) isOnewayMethod.invoke(entry.getValue())) {
+          onewayMethods.add(entry.getKey());
+        }
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+    log.debug("Found oneway Thrift methods: " + onewayMethods);
 
-  public static <T> T service(final T instance) {
     InvocationHandler handler = new RpcServerInvocationHandler<T>(instance) {
       private final Logger log = LoggerFactory.getLogger(instance.getClass());
 
       @Override
       public Object invoke(Object obj, Method method, Object[] args) throws Throwable {
+        // e.g. ThriftClientHandler.flush(TInfo, TCredentials, ...)
         try {
           return super.invoke(obj, method, args);
         } catch (RuntimeException e) {
+          if (onewayMethods.contains(method.getName())) {
```

Comment:
> I believe that Thrift disallows multiple methods with the same name in one service

Yes, this is the case. Each Thrift service (e.g. master, client, tabletserver) will have its own instance of RpcWrapper underneath the `Processor`. Thus, the methods contained are unique to a service definition whose method names are guaranteed to be unique.